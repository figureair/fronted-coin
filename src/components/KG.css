#all{
    background-size: 100% auto;
    height: 100%;
    position: absolute;
    width: 100%;
    display: flex;
    flex-direction: row;
}

#main{
    background-size: 100% auto;
    height: 100%;
    background: #eeeff1;
    width: 80%;
    display: flex;
    justify-content: center;
    align-items: center;
}
.box {
    display: flex;
    width: 80%;
    height: 80%;
    background: white;
    border: 1px solid rgba(140, 138, 138, 0.25);
    border-radius: 10px;
    box-shadow: 0 2px 4px rgba(0, 0, 0, .12), 0 0 6px rgba(0, 0, 0, .04);

    -webkit-animation: shadow-drop-2-center 0.4s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;
    animation: shadow-drop-2-center 0.4s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;
}

#myChart {
    width: 100%;
    height: 100%;
}

.options {
    top: 20%;
    display: flex;
    flex-direction: column;
    align-items: center;
    position: fixed;
    right: 3%;
}

.option {
    /*position: fixed;*/
    height: 40px;
    margin-left: 20px;
}

#text-box {
    width: 24%;
    height: 80vh;
    border: 1px solid rgba(140, 138, 138, 0.25);
    border-radius: 10px;
    margin-left: 20px;
    text-align: center;
    box-shadow: 0 2px 4px rgba(0, 0, 0, .12), 0 0 6px rgba(0, 0, 0, .04)
}

#recommend{
    width: 100%;
    height: 100%;
    text-align: center;
    display: flex;
    flex-direction: row;
}

.recommend_item{
    flex:1;
    display: flex;
    flex-direction: column;
    justify-content: center; /* 水平居中 */
    align-items: center;     /* 垂直居中 */
}

.recommend_list_items{
    height:25vh;
    width:90%;
    overflow: auto;
}

.recommend_title{
    height: 10vh;
}

.recommend_bottom{
    height: 5vh;
    float: right;
}

#user_pic{
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: row;
}

#user_pic_box1{
    width:30%;
    height:100%;
    display: flex;
    flex-direction: column;
}

#user_pic_box1_item1{
    width: 100%;
    height: 10vh;
    font-size: 3vh;
    font-weight: bold;
}

.user_pic_box1_item2{
    width: 100%;
    height: 30vh;
    font-size: 20px;
}

#user_pic_box2{
    width:70%;
    height:100%;
    display: flex;
    flex-direction: row;
    flex-flow: wrap;

}

.user_pic_box2_item{
    width:30vh;
    height:30vh;

}

.four{
    width: 50%;
    height: 50%;
    margin-top: 2vh;
}

#person_info{
    width: 95.5%;
    height: 80vh;
    margin-left: 30px;
    text-align: center;
    display: flex;
    flex-direction: row;
}

#person_info_box1{
    width:65%;
    height:80vh;
}

#simple_person_info{
    width: 100%;
    height:10vh;
    margin-top: -2vh;
    /*border:1px solid black;*/
}

#movie_list{
    width:100%;
    height:70vh;
    margin-top: -2vh;
    display: flex;
    flex-direction: row;
}

.movie_list_item{
    flex:1;
    display: flex;
    flex-direction: column;
    justify-content: center; /* 水平居中 */
    align-items: center;     /* 垂直居中 */
}

.movie_list_items{
    height:80%;
    width:90%;
    overflow: auto;
}


#person_info_box2{
    width:35%;
    height:80vh;
    display: flex;
    flex-direction: column;
}

#person_pic{
    width: 70vh;
    height: 80vh;
}

#movie_info{
    width: 95.5%;
    height: 40vh;
    margin-left: 30px;
    text-align: center;
    display: flex;
    flex-direction: column;
}

#movie_info_box1{
    width:100%;
    height:10vh;
}

#movie_info_box2{
    width:100%;
    height:30vh;
    display: flex;
    flex-direction: row;
}

.movie_info_item{
    flex: 1;
}

.love_button{
    display: flex;
    flex-direction: row;
    justify-content: center;
}

.box-item {
    display: flex;
    flex-direction: row;
    width: 100%;
    justify-content: space-evenly;
    text-align: center;
    align-items: center;
    margin-top: 10px;
}

#selector-box {
    justify-content: center;
}

#selector-title {
    text-align: center;
    line-height: 40px;
    margin-right: 10px;
    overflow: visible;
}

.form {
    max-width: 400px;
    max-height: 280px;
    border: 1px;
}

#checkbox1 {
    margin-top: 10px;
    margin-right: 10px;
}

.detail-info {
    font-size: 14px;
    text-align: center;
}

.searchInput {
    width: 70%;
    margin: 0 10px 0;
}

.download-box{
    margin-top: 50px;
}

.download-box-item{
    margin-top: 20px;
}

.style-box-item{
    width: 100%;
    margin: 10px 10px 0;
}

.search-box-item{

    width: 100%;
    justify-content: space-evenly;
    text-align: center;
    align-items: center;

    overflow-y: auto;
    height:500px;
    margin-top: 30px;
}

.search-form{
    margin-left: 30px;
    margin-right: 30px;
}

.el-form{
    max-height: 600px;
    overflow: auto;
}

.edit-button {
    font-size: 120%;
}

.el-button-group>.el-button {
    float: none;
    position: relative;
}

#menu-content {
    /*width: 100%;*/

}

.like-text {
    font-size: 20px;
    width: 100%;
}

.like {
    flex-wrap: wrap;
}

.heart {
    margin: 17px;
}

#chat {
    position: absolute;
    left: 120px;
    top: 89px;
}

.recommendImg{
    max-width: 80%;
    max-height: 80%;
    display: block;
    margin: auto;
}
#bottom{
    max-height: 40vh;
}

.el-menu {
    display: flex;
}

#logout {
    margin-left: auto;
}

.inside-box {
    margin-left: 10px;
    margin-right: 10px;
}

#editOpts {
    display: flex;
    flex-direction: column;
}

.view {
    display: flex;
    align-items: center;
    flex-direction: column;
    margin: 0 10px;
}

#sidebar{
    background: #181717;
    width: 20%;
    height: 100%;
}

#system_title {
    height: 20%;
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: center;
}

#system_title_text {
    height: 100%;
    width: 55%;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

#system_title_pic {
    height: 100%;
    width: 45%;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

#movieImg {
    max-width: 50%;
    max-height: 50%;
    display: block;
    margin: auto;
}

.system_title_text_box {
    display: flex;
    flex-direction: row;
    color: #f5f5f5;
    font-size: 28px;
    font-weight: bold;
}

#system_tools {
    height: 50%;
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: center;
}

#tools_box{
    height: 100%;
    width: 80%;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
}

.system_tools_title{
    height: 8%;
    margin-left: 20px;
    display: flex;
    flex-direction: row;
    color: #f5f5f5;
    font-size: 20px;
    margin-top: 30px;
}

.system_tools_box{
    height: 10%;
    width: 100%;
    margin-top: 20px;
    display: flex;
    flex-direction: row;
    border-radius:10px;
}

.system_tools_box:hover
{
    background: #454444;
}

.system_tools_box_primary{
    height: 10%;
    width: 100%;
    margin-top: 20px;
    display: flex;
    flex-direction: row;
    border-radius:10px;
    background: #454444;
}

.toolsImg{
    margin-left: 20px;
    max-width: 100%;
    max-height: 100%;
    display: block;
}

.system_tools_text_box{
    height: 100%;
    width: 70%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    color: #f5f5f5;
    font-size: 25px;
}

.system_tools_pic{
    height: 100%;
    width: 20%;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

#QA_box{
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

#QA_inner_box{
    width: 50%;
    height: 70%;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.mini_box{
    position: absolute;
    bottom: 1vh;
    right: 1vh;
    width: 20%;
    height: 30%;
    border: 1px solid rgba(140, 138, 138, 0.25);
    border-radius: 10px;
    box-shadow: 0 2px 4px rgba(0, 0, 0, .12), 0 0 6px rgba(0, 0, 0, .04);
    background: white;
}

#recommend_exit{
    position: absolute;
    left: 1vh;
    top:1vh;
}

.mini_button{
    position: absolute;
    bottom: 1vh;
    right: 1vh;
    background: white;
}

#chatImg{
    max-width: 100%;
    max-height: 100%;
    display: block;
    margin: auto;
}

#conImg{
    max-width: 100%;
    max-height: 100%;
    display: block;
    margin: auto;
}

@-webkit-keyframes shadow-drop-2-center {
    0% {
        -webkit-transform: translateZ(0);
        transform: translateZ(0);
        -webkit-box-shadow: 0 0 0 0 rgba(0, 0, 0, 0);
        box-shadow: 0 0 0 0 rgba(0, 0, 0, 0);
    }
    100% {
        -webkit-transform: translateZ(50px);
        transform: translateZ(50px);
        -webkit-box-shadow: 0 0 20px 0px rgba(0, 0, 0, 0.35);
        box-shadow: 0 0 20px 0px rgba(0, 0, 0, 0.35);
    }
}
@keyframes shadow-drop-2-center {
    0% {
        -webkit-transform: translateZ(0);
        transform: translateZ(0);
        -webkit-box-shadow: 0 0 0 0 rgba(0, 0, 0, 0);
        box-shadow: 0 0 0 0 rgba(0, 0, 0, 0);
    }
    100% {
        -webkit-transform: translateZ(50px);
        transform: translateZ(50px);
        -webkit-box-shadow: 0 0 20px 0px rgba(0, 0, 0, 0.35);
        box-shadow: 0 0 20px 0px rgba(0, 0, 0, 0.35);
    }
}

