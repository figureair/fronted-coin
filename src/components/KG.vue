<template>
  <div id="box">
    <div id="myChart">
    </div>
    <div id="text-box">
      <select id="selector" @change="changeTo()">
        <option>关系图</option>
        <option>力引导图</option>
        <option>环形关系图</option>
      </select>
    </div>
  </div>
</template>


<script>
import $ from 'jquery'

let myChart;
let savedgraph;
const ROOT_PATH = 'https://cdn.jsdelivr.net/gh/apache/echarts-website@asf-site/examples';

//目前存有三种模式，后续迭代将加入更多表现模式
//1.关系图
let option1;
//2.力引导图
let option2;
//3.环形关系图
let option3;

export default {
  name: "KG",
  mounted() {
    this.drawLine();
  },
  methods: {
    drawLine() {
      // 初始化echarts实例
      myChart = this.$echarts.init(document.getElementById('myChart'))
      // 绘制图表
      myChart.showLoading();
      $.getJSON(ROOT_PATH + '/data/asset/data/les-miserables.json', function (graph) {
        myChart.hideLoading();
        //保存原始数据
        savedgraph = JSON.parse(JSON.stringify(graph))

        //初始设置为option1
        graph.nodes.forEach(function (node) {
          node.label = {
            show: node.symbolSize > 30
          };
        });
        option1 = {
          tooltip: {
            position: 'right',
            extraCssText: 'box-shadow: 0 0 3px rgba(0, 0, 0, 0.3)'
          },
          //图例
          legend: [{
            data: graph.categories.map(function (a) {
              return a.name;
            })
          }],
          animationDuration: 1500,
          animationEasingUpdate: 'quadraticIn',
          series: [
            {
              type: 'graph',
              //不采用任何布局
              layout: 'none',
              //关闭悬停图例高亮
              legendHoverLink: false,
              //节点大小不随鼠标缩放而缩放
              nodeScaleRatio: 0,
              //开启鼠标缩放和漫游
              roam: true,
              //边两端的标记
              edgeSymbol: ['none', 'arrow'],
              //边两端的标记大小
              edgeSymbolSize: 5,
              //悬停时鼠标样式
              cursor: 'pointer',

              data: graph.nodes,
              links: graph.links,
              categories: graph.categories,

              label: {
                position: 'right',
                formatter: '{b}'
              },

              lineStyle: {
                color: 'source',
                curveness: 0.3
              },

              emphasis: {
                scale: true,
                focus: 'adjacency',
                lineStyle: {
                  width: 10
                }
              }
            }
          ]
        };
        myChart.setOption(option1);

        //预存option2
        graph = JSON.parse(JSON.stringify(savedgraph))
        graph.nodes.forEach(function (node) {
          node.label = {
            show: node.symbolSize > 30
          };
          node.symbolSize = 5;
        });
        option2 = {
          tooltip: {
            position: 'right',
            extraCssText: 'box-shadow: 0 0 3px rgba(0, 0, 0, 0.3)'
          },
          //图例
          legend: [{
            data: graph.categories.map(function (a) {
              return a.name;
            })
          }],
          series: [
            {
              //当前视角的缩放比例
              zoom: 2,
              //是否可拖动
              draggable: true,
              type: 'graph',
              //类型为力引导图
              layout: 'force',

              data: graph.nodes,
              links: graph.links,
              categories: graph.categories,

              //开启鼠标缩放和漫游
              roam: true,
              label: {
                position: 'right'
              },
              force: {
                //斥力
                repulsion: 100
              }
            }
          ]
        };

        //预存option3
        graph = JSON.parse(JSON.stringify(savedgraph))
        graph.nodes.forEach(function (node) {
          node.label = {
            show: node.symbolSize > 30
          };
        });
        option3 = {
          tooltip: {
            position: 'right',
            extraCssText: 'box-shadow: 0 0 3px rgba(0, 0, 0, 0.3)'
          },
          //图例
          legend: [{
            data: graph.categories.map(function (a) {
              return a.name;
            })
          }],
          animationDurationUpdate: 1500,
          animationEasingUpdate: 'quinticInOut',
          series: [
            {
              type: 'graph',
              layout: 'circular',
              circular: {
                rotateLabel: true
              },

              data: graph.nodes,
              links: graph.links,
              categories: graph.categories,

              roam: true,

              label: {
                position: 'right',
                formatter: '{b}'
              },
              lineStyle: {
                color: 'source',
                curveness: 0.3
              },
              emphasis: {
                focus: 'adjacency',
                lineStyle: {
                  width: 10
                }
              }
            }
          ]
        };
      });
    },

    changeTo() {
      let index = document.getElementById("selector").selectedIndex + 1;
      console.log(index)
      myChart.clear();
      switch (index) {
        case 1:
          myChart.setOption(option1);
          break;
        case 2:
          myChart.setOption(option2);
          break;
        case 3:
          myChart.setOption(option3);
          break;
      }
    }

  }
}
</script>

<style scoped>
#box {
  display: flex;
  width: 100%;

}

#myChart {
  width: 70%;
  height: 80vh;
  border: 1px solid rgba(200, 200, 200, 0.75);
  border-radius: 10px;
  margin-left: 30px;
}

#text-box {
  width: 24%;
  height: 80vh;
  border: 1px solid rgba(200, 200, 200, 0.75);
  border-radius: 10px;
  margin-left: 20px;
}

</style>