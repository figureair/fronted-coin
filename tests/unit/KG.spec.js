import {mount, createLocalVue, shallowMount} from '@vue/test-utils'
import KG from '@/components/KG.vue'
import ElementUI from 'element-ui';
import * as echarts from 'echarts';

const localVue = createLocalVue()
localVue.use(ElementUI)

let tmpdom = document.createElement('div')
Object.defineProperty(tmpdom, 'clientHeight', {
    get: function () {
        return 500;
    }
})
Object.defineProperty(tmpdom, 'clientWidth', {
    get: function () {
        return 500;
    }
})

describe('KG.vue', () => {

    it('快照测试，渲染正确UI', () => {
        const wrapper = shallowMount(KG, {localVue})
        expect(wrapper).toMatchSnapshot()
        wrapper.destroy()
    })

    it('点击导入须知，弹出dialog，点击确认关闭', () => {
        const wrapper = mount(KG, {localVue})
        wrapper.find('#tip').trigger('click')
        expect(wrapper.vm.dialogVisible).toBe(true)
        wrapper.find('#tipclose').trigger('click')
        expect(wrapper.vm.dialogVisible).toBe(false)
        wrapper.destroy()
    })

    it('点击下载Img，触发downloadImg', async () => {

        const wrapper = mount(KG, {localVue})

        wrapper.setData({
            myChart: echarts.init(tmpdom),
            savedgraph: JSON.parse('{"nodes":[{"symbolSize": 40,"name": "当事人","category": 0,"x":"1","y":"2"},{"symbolSize": 40,"name": "123","category": 0,"x":"2","y":"1"}],"links":[{"source": "0","target": "1"}],"categories":[{"name": "Person"}]}')
        })
        const spyFn = jest.spyOn(wrapper.vm, 'downloadImg')
        await wrapper.vm.$forceUpdate();

        //mock getDataURL方法 来自定义返回
        wrapper.vm.myChart.getDataURL = function () {
            return "123"
        }

        wrapper.find('#downimg').trigger('click')
        expect(spyFn).toHaveBeenCalled()

        echarts.dispose(tmpdom)
        wrapper.destroy()
    })

    it('点击下载Json，触发downloadJson', async () => {
        const wrapper = mount(KG, {localVue})
        const spyFn = jest.spyOn(wrapper.vm, 'downloadJson')
        await wrapper.vm.$forceUpdate();
        wrapper.find('#downjson').trigger('click')
        expect(spyFn).toHaveBeenCalled()
        wrapper.destroy()
    })

    it('点击下载Xml，触发downloadXml', async () => {
        const wrapper = mount(KG, {localVue})
        wrapper.setData({
            savedgraph: JSON.parse('{"nodes":[{"symbolSize": 40,"name": "当事人","category": 0,"x":"1","y":"2"},{"symbolSize": 40,"name": "123","category": 0,"x":"2","y":"1"}],"links":[{"source": "0","target": "1"}],"categories":[{"name": "Person"}]}')
        })
        const spyFn = jest.spyOn(wrapper.vm, 'downloadXml')
        if (typeof window.URL.createObjectURL === 'undefined') {
            window.URL.createObjectURL = () => {

            };
        }
        await wrapper.vm.$forceUpdate();
        wrapper.find('#downxml').trigger('click')
        expect(spyFn).toHaveBeenCalled()
        wrapper.destroy()
    })

    it('导入非json文件，提示错误', async () => {
        const wrapper = mount(KG, {localVue})
        let file = new File([""], "123.txt", {type: "text/plain"});
        expect(wrapper.vm.beforeJSONUpload(file)).toBe(false)
        wrapper.destroy()
    })

    it('导入json文件，跳转', async () => {
        const wrapper = mount(KG, {localVue})
        let file = new File([""], "123", {type: "application/json"});
        expect(wrapper.vm.beforeJSONUpload(file)).toBe(true)
        wrapper.destroy()
    })

    it("文件格式错误，提示错误", () => {
        const wrapper = mount(KG, {localVue})
        expect(wrapper.vm.checkjson(JSON.parse('{}'))).toBe(false)
        expect(wrapper.vm.checkjson(JSON.parse('{"nodes":[]}'))).toBe(false)
        expect(wrapper.vm.checkjson(JSON.parse('{"nodes":[],"links":[]}'))).toBe(false)
        expect(wrapper.vm.checkjson(JSON.parse('{"nodes":[],"links":[],"categories":[]}'))).toBe(false)
        expect(wrapper.vm.checkjson(JSON.parse('{"nodes":[{"symbolSize": 40,"name": "当事人","category": 0}],"links":[{"source": "3"}],"categories":[{"name": "Person"}]}'))).toBe(false)
        expect(wrapper.vm.checkjson(JSON.parse('{"nodes":[{"name": "当事人","category": 0}],"links":[{"source": "3","target": "0"}],"categories":[{"name": "Person"}]}'))).toBe(false)
        expect(wrapper.vm.checkjson(JSON.parse('{"nodes":[{"symbolSize": 40,"name": "当事人","category": 0}],"links":[{"source": "3","target": "0"}],"categories":[{}]}'))).toBe(false)
        wrapper.destroy()
    })

    it("文件正确格式，返回true", async () => {
        const wrapper = mount(KG, {localVue})
        expect(wrapper.vm.checkjson(JSON.parse('{"nodes":[{"symbolSize": 40,"name": "当事人","category": 0}],"links":[{"source": "3","target": "0"}],"categories":[{"name": "Person"}]}'))).toBe(true)
        expect(wrapper.vm.checkjson(JSON.parse('{"nodes":[{"symbolSize": 40,"name": "当事人","category": 0,"x":"1","y":"2"}],"links":[{"source": "3","target": "0"}],"categories":[{"name": "Person"}]}'))).toBe(true)
        wrapper.destroy()
    })

    it('选择样式，测试触发', async () => {
        const wrapper = mount(KG, {localVue})
        wrapper.setData({
            myChart: echarts.init(tmpdom),
            option1: {},
            option2: {},
            option3: {}
        })
        expect(wrapper.vm.changeTo(1)).toEqual(1);
        expect(wrapper.vm.changeTo(2)).toEqual(2);
        expect(wrapper.vm.changeTo(3)).toEqual(3);

        echarts.dispose(tmpdom)
        wrapper.destroy()
    })

    it('配置渲染，测试触发', async () => {
        const wrapper = mount(KG, {localVue})
        wrapper.setData({
            changeOption:jest.fn(function() {
                return true;
            }),
            myChart: echarts.init(tmpdom),
            savedgraph: JSON.parse('{"nodes":[{"symbolSize": 40,"name": "当事人","category": 0,"x":"1","y":"2"},{"symbolSize": 40,"name": "123","category": 0,"x":"2","y":"1"}],"links":[{"name":"dot","source": "0","target": "1"}],"categories":[{"name": "Person"}]}')
        })
        const spyFn = jest.spyOn(wrapper.vm, "initpage");
        wrapper.vm.initpage()
        expect(spyFn).toHaveBeenCalled()
        echarts.dispose(tmpdom)
        wrapper.destroy()
    })

    it("initdata初始化测试", async () => {
        const wrapper = mount(KG, {localVue})
        const spyFn = jest.spyOn(wrapper.vm, "initdata");
        await wrapper.vm.$forceUpdate();
        wrapper.vm.initdata()
        expect(spyFn).toHaveBeenCalled()
        wrapper.destroy()
    })

    it("changeOption测试", () => {
        const wrapper = mount(KG, {localVue})
        wrapper.setData({
            myChart: echarts.init(tmpdom)
        })
        expect(wrapper.vm.changeOption({})).toBe(true)
        echarts.dispose(tmpdom)
        wrapper.destroy()
    })

    it("删除节点", () => {
        const wrapper = mount(KG, {localVue})
        wrapper.setData({
            initpage:jest.fn(function() {
                return true;
            }),
            savedgraph: JSON.parse('{"nodes":[{"symbolSize": 40,"name": "当事人","category": 0,"x":"1","y":"2"},{"symbolSize": 40,"name": "123","category": 0,"x":"2","y":"1"}],"links":[{"name":"dot","source": "0","target": "1"}],"categories":[{"name": "Person"}]}')
        })
        const spyFn = jest.spyOn(wrapper.vm, "deleteNode");
        wrapper.vm.deleteNode({index:1})
        expect(spyFn).toHaveBeenCalled()
        wrapper.destroy()
    })

    it("删除边", () => {
        const wrapper = mount(KG, {localVue})
        wrapper.setData({
            initpage:jest.fn(function() {
                return true;
            }),
            savedgraph: JSON.parse('{"nodes":[{"symbolSize": 40,"name": "当事人","category": 0,"x":"1","y":"2"},{"symbolSize": 40,"name": "123","category": 0,"x":"2","y":"1"}],"links":[{"name":"dot","source": "0","target": "1"}],"categories":[{"name": "Person"}]}')
        })
        const spyFn = jest.spyOn(wrapper.vm, "deleteEdge");
        wrapper.vm.deleteEdge({index:1})
        expect(spyFn).toHaveBeenCalled()
        wrapper.destroy()

    })

    it("增加节点，增加边", () => {
        const wrapper = mount(KG, {localVue})
        wrapper.setData({
            initpage:jest.fn(function() {
                return true;
            }),
            cleanField:jest.fn(function() {
                return true;
            }),
            savedgraph: JSON.parse('{"nodes":[{"symbolSize": 40,"name": "当事人","category": 0,"x":"1","y":"2"},{"symbolSize": 40,"name": "123","category": 0,"x":"2","y":"1"}],"links":[{"name":"dot","source": "0","target": "1"}],"categories":[{"name": "Person"}]}'),
            addNodeForm:{category:1,name:"123",symbolSize:50,id:3,value:"1",target:1,source:2}
        })
        wrapper.vm.isValid(true,'node')
        wrapper.vm.isValid(true,'edge')
        wrapper.vm.cancel('edge')
        wrapper.vm.cancel('node')
        wrapper.destroy()
    })

    it("校验方法测试", () => {
        const wrapper = mount(KG, {localVue})
        const mockFn=jest.fn(function() {
            return true;
        })
        wrapper.setData({
            selectedItem:[{index:1}],
            savedgraph: JSON.parse('{"nodes":[{"id":1,"symbolSize": 40,"name": "当事人","category": 0,"x":"1","y":"2"},{"symbolSize": 40,"name": "123","category": 0,"x":"2","y":"1"}],"links":[{"name":"dot","source": "0","target": "1"}],"categories":[{"name": "Person"}]}')
        })
        wrapper.vm.checkName('','当事人',mockFn)
        wrapper.vm.checkName('','',mockFn)
        wrapper.vm.checkSymbolSize('','',mockFn)
        wrapper.vm.checkSymbolSize('',-1,mockFn)
        wrapper.vm.checkSymbolSize('',1,mockFn)
        wrapper.vm.checkSymbolSize('',NaN,mockFn)
        wrapper.vm.checkTarget('','',mockFn)
        wrapper.vm.checkSource('','',mockFn)
        wrapper.vm.checkCategory('','',mockFn)
        wrapper.vm.checkCategory('',NaN,mockFn)
        wrapper.vm.checkCategory('',-1,mockFn)
        wrapper.vm.checkCategory('',10,mockFn)
        wrapper.vm.checkCategory('',0,mockFn)
        wrapper.vm.checkEditSource('','',mockFn)
        wrapper.vm.checkEditSource('',10,mockFn)
        wrapper.vm.checkEditSource('',1,mockFn)
        wrapper.vm.checkEditTarget('','',mockFn)
        wrapper.vm.checkEditTarget('',10,mockFn)
        wrapper.vm.checkEditTarget('',1,mockFn)
        wrapper.vm.checkEditName('','',mockFn)
        wrapper.vm.checkEditName('','当事人',mockFn)
        wrapper.vm.checkEditName('','123',mockFn)
        wrapper.destroy()
    })

    it("修改节点，修改边", () => {
        const wrapper = mount(KG, {localVue})
        wrapper.setData({
            initpage:jest.fn(function() {
                return true;
            }),
            checked:true,
            selectedItem:[{}],
            editable:true,
            input:{index:0,source:1,target:2,name:'123',value:1,symbolSize:30},
            savedgraph: JSON.parse('{"nodes":[{"symbolSize": 40,"name": "当事人","category": 0,"x":"1","y":"2"},{"symbolSize": 40,"name": "123","category": 0,"x":"2","y":"1"}],"links":[{"name":"dot","source": "0","target": "1"}],"categories":[{"name": "Person"}]}')
        })
        const spyFn = jest.spyOn(wrapper.vm, "checkValidEdit");
        wrapper.vm.showAllInfo()
        expect(wrapper.vm.editable).toBe(false)
        wrapper.vm.checkValidEdit(true,'edge')
        wrapper.vm.checkValidEdit(true,'node')
        expect(spyFn).toHaveBeenCalled()

        wrapper.destroy()
    })

    it("修改时传递参数", () => {
        const wrapper = mount(KG, {localVue})
        wrapper.vm.startEdit({index:0,source:1,target:2},'edge')
        expect(wrapper.vm.input).toEqual({index:0,source:1,target:2})
        wrapper.vm.startEdit({index:0,id:1,category:1,name:'123',value:1,symbolSize:30},'node')
        expect(wrapper.vm.input).toEqual({index:0,id:1,category:1,name:'123',value:1,symbolSize:30})

        wrapper.destroy()
    })
})