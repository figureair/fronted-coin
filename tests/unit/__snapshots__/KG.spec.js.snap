// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KG.vue 快照测试，渲染正确UI 1`] = `
<div id="box">
  <div id="myChart"></div>
  <div id="text-box">
    <div id="selector-box" class="box-item">
      <div id="selector-title">样式选择:</div>
      <el-select-stub id="selector" value="" autocomplete="off" multiplelimit="0" valuekey="value" popperappendtobody="true">
        <el-option-stub value="1" label="关系图"></el-option-stub>
        <el-option-stub value="2" label="力引导图"></el-option-stub>
        <el-option-stub value="3" label="环形关系图"></el-option-stub>
      </el-select-stub>
    </div>
    <div class="box-item">
      <el-button-stub type="primary" icon="el-icon-download" nativetype="button" plain="true" id="downimg">下载Img</el-button-stub>
    </div>
    <div class="box-item">
      <el-button-stub type="primary" icon="el-icon-download" nativetype="button" plain="true" id="downjson">下载Json</el-button-stub>
    </div>
    <div class="box-item">
      <el-button-stub type="primary" icon="el-icon-download" nativetype="button" plain="true" id="downxml">下载Xml</el-button-stub>
    </div>
    <div class="box-item ">
      <el-upload-stub action="" headers="[object Object]" name="file" showfilelist="true" type="select" beforeupload="function () { [native code] }" onremove="function srcvue_type_script_lang_js_noop() {}" onchange="function srcvue_type_script_lang_js_noop() {}" onsuccess="function srcvue_type_script_lang_js_noop() {}" onprogress="function () { [native code] }" onerror="function srcvue_type_script_lang_js_noop() {}" filelist="" autoupload="true" listtype="text" onexceed="function srcvue_type_script_lang_js_noop() {}">
        <el-button-stub type="primary" icon="el-icon-upload" nativetype="button" plain="true" id="upload_button">导入知识图谱</el-button-stub>
      </el-upload-stub>
      <el-button-stub type="text" icon="" nativetype="button" id="tip">导入须知</el-button-stub>
      <el-dialog-stub modal="true" modalfade="true" modalappendtobody="true" lockscroll="true" closeonpressescape="true" closeonclickmodal="true" title="导入须知" showclose="true" width="30%" customclass="" top="15vh"><span>目前只支持json文件。<br>json对象中必须包含nodes，links，categories三个属性。<br>每一个node须包含name，symbolSize，category属性。<br>每一个link须包含source，target属性。<br>每一个category须包含name属性。
        </span> <span class="dialog-footer"><el-button-stub type="primary" icon="" nativetype="button" id="tipclose">确 定</el-button-stub></span></el-dialog-stub>
    </div>
    <el-popover-stub transformorigin="true" placement="left" boundariespadding="5" offset="0" visiblearrow="true" arrowoffset="0" appendtobody="true" popperoptions="[object Object]" trigger="click" opendelay="0" closedelay="200" transition="fade-in-linear" tabindex="0">
      <!---->
      <!---->
    </el-popover-stub>
    <div class="box-item">
      <el-button-stub type="primary" icon="" nativetype="button" plain="true">查 看 数 据</el-button-stub>
    </div>
    <div class="box-item">
      <el-popover-stub transformorigin="true" placement="left" boundariespadding="5" offset="0" visiblearrow="true" arrowoffset="0" appendtobody="true" popperoptions="[object Object]" trigger="click" opendelay="0" closedelay="200" width="400px" transition="fade-in-linear" tabindex="0" class="form">
        <!---->
      </el-popover-stub>
      <el-button-stub type="primary" icon="" nativetype="button" plain="true">添加节点</el-button-stub>
      <el-popover-stub transformorigin="true" placement="left" boundariespadding="5" offset="0" visiblearrow="true" arrowoffset="0" appendtobody="true" popperoptions="[object Object]" trigger="click" opendelay="0" closedelay="200" width="400px" transition="fade-in-linear" tabindex="0" class="form">
        <!---->
      </el-popover-stub>
      <el-button-stub type="primary" icon="" nativetype="button" plain="true">添加边</el-button-stub>
    </div>
  </div>
</div>
`;
